Format name accordingly

Chairn opened the pull request:
I think i got almost all the misstyped name. I did let some because they made sense to me. I think, only generated code remains to be formatted (and src/base and websockets, but those seems to have an exception).

## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [x] Changed no physics that affect existing maps
- [x] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

heinrich5991 commented:
Nice, how did you manage to do this?

Learath2 commented:
> Nice, how did you manage to do this?

The regexes he used are in the commit messages. Which concerns me a little, parsing C++ with an FSA sounds dangerous

Chairn commented:
I used regex to find those, but i replaced them either manually at start, then it got boring so i used replace all reference of VSCode.

Chairn commented:
Okay, ready to merge. I fixed macos compilation.

I have found one variable that i replaced erroneously (corrected in the last commit with regex included in the commit message) and 2 misnamed non pointer variable as pointer. So i guess (hope) i made no mistake.

heinrich5991 commented on src/engine/shared/masterserver.cpp:
```diff
@@ -1,12 +1,12 @@
-const unsigned char SERVERBROWSE_GETINFO[SERVERBROWSE_SIZE] = {255, 255, 255, 255, 'g', 'i', 'e', '3'};
-const unsigned char SERVERBROWSE_INFO[SERVERBROWSE_SIZE] = {255, 255, 255, 255, 'i', 'n', 'f', '3'};
+const unsigned char g_aServerBrowseGetInfo[SERVERBROWSE_SIZE] = {255, 255, 255, 255, 'g', 'i', 'e', '3'};
+const unsigned char g_aServerBrowseInfo[SERVERBROWSE_SIZE] = {255, 255, 255, 255, 'i', 'n', 'f', '3'};
```
We previously allowed (or forced?) global constants to have `SCREAMING_SNAKE_CASE` names. Do we want to change that?

Chairn replied:
I put these specifically in a separate commit for easier revert ;)

heinrich5991 commented:
Oh damn, lots of conflicts. :( Could you rebase this without the globals for so that we can merge this and it doesn't accumulate more conflicts?

Chairn commented:
Do you want me to also remove https://github.com/ddnet/ddnet/pull/5545/commits/fd63ebcfc6c49a3571fb03c03a2a97d4cabe1ac9 which concerns test only?

heinrich5991 commented:
Yes, all the "changing away from `SCREAMING_SNAKE_CASE` constants".

Chairn commented on src/engine/client/updater.cpp:
```diff
@@ -62,19 +62,19 @@ int CUpdaterFetchTask::OnCompletion(int State)
-			b = a + 1;
-	b = b ? b : Dest();
-	if(!str_comp(b, "update.json"))
+	const char *pThanksForTheMeaningVariableName = 0;
+	for(const char *pThanksAgain = Dest(); *pThanksAgain; pThanksAgain++)
+		if(*pThanksAgain == '/')
+			pThanksForTheMeaningVariableName = pThanksAgain + 1;
+	pThanksForTheMeaningVariableName = pThanksForTheMeaningVariableName ? pThanksForTheMeaningVariableName : Dest();
+	if(!str_comp(pThanksForTheMeaningVariableName, "update.json"))
```
Here and above, i had no idea what the variable meant. I even forgot the ful in meaningful

Chairn commented on src/game/client/components/controls.cpp:
```diff
@@ -85,15 +85,15 @@ static void ConKeyInputCounter(IConsole::IResult *pResult, void *pUserData)
+
+	// if(((*v) & 1) != pResult->GetInteger(0))
+	// 	(*v)++;
+	// *v &= INPUT_STATE_MASK;
```
Dead code (i commented it)?

def- replied:
No, doesn't look dead. It changes pState->m_pVariable[12]

Chairn commented:
Okay, i got all the snakes that tried to sneak on me :)

heinrich5991 commented:
Can I squash the commits so we don't have many formatting-related commits?

Chairn commented:
Sure i'll do it right now

heinrich5991 commented:
bors r+

Thanks! :)

def- commented:
I'm not sure if these kind of changes are a good idea. Every revert is extremely painful now, everyone maintaing mods will have a lot of pain with this too.

C0D3D3V commented:
If you mean with diff also git blame... What I do about this is the following: The last big format was this https://github.com/ddnet/ddnet/commit/3be8a592e52a03e555b6aa29a8dea529797bb645
You can see that basically everywhere in git-blame that commit. If I want to know what was the change before the formatting was I can just use the shortcut to "Open Line Changes with Previous Revision" and then I see the commit message from before. I can repeat that multiple times. Like this:
![grafik](https://user-images.githubusercontent.com/14315968/178137299-323bf25c-6c02-4578-be17-7f38a3e62e5d.png)
 
 
 I think such formatting commits to force our formatting rules on everything is necessary, else we will never have all rules applied, and devs (like me) trying to avoid them, because they have the argument it is nowhere applied, so it is not really a rule. 
 
 I guess for other mods that means just one new commit to apply all the rules to there code (at least for all our variables). Should be easily be done by try compile and error.

Chairn commented:
> I'm not sure if these kind of changes are a good idea. Every revert is extremely painful now, everyone maintaing mods will have a lot of pain with this too. And every diff is just showing that Chairn wrote everything, so diffs are also not that useful. (I'll change the formatting-revs.txt for the last point, but can't do anything about the first two)

As said by C0D3D3V, it is necessary (imo) to enforce the correct use of rules for everyone. Note that this should be close to the last big formatting. I just have double pointers that i have forgotten in this PR, the case being discussed in https://github.com/ddnet/ddnet/issues/5569, protocol files that are generated and everything in `src/base` + `websockets.cpp`.
